backend/scrapers/em2.js

```diff
@@ -0,0 +1,371 @@
+    type: 'db-request',
+    subtype: 'isRefused',
+    messageData: {
+      string: event.title,
+    },
+  });
+  await this.checkDBhasAnswered();
+  reasons.push(this.lastDBAnswer.reason);
+  if (this.lastDBAnswer.success) {
+    return {
+      event,
+      reason: reasons.reverse().join(','),
+      success: false,
+    };
+  }
+
+  const hasForbiddenTerms = await this.hasForbiddenTerms(event);
+  reasons.push(hasForbiddenTerms.reason);
+  if (hasForbiddenTerms.success) {
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveRefusedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    });    
+    
+    return {
+      event,
+      reason: reasons.reverse().join(','),
+      success: false,
+    };
+  }
+
+  const isRockRes = await this.isRock(event);
+  if (isRockRes.success) {
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveAllowedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    }); 
+    return {
+      event,
+      reason: reasons.reverse().join(', '),
+      success: true,
+    };
+  }
+
+  const goodTermsRes = await this.hasGoodTerms(event);
+  reasons.push(goodTermsRes.reason);
+  if (goodTermsRes.success) {
+    this.skipFurtherChecks.push(event.title);
+    this.talkToDB({
+      type: 'db-request',
+      subtype: 'saveAllowedTitle',
+      messageData: {
+        string: event.title,
+        reason: reasons.reverse().join(', '),
+      },
+    });
+    return goodTermsRes;
+  }
+
+
+  this.talkToDB({
+    type: 'db-request',
+    subtype: 'saveRefusedTitle',
+    messageData: {
+      string: event.title,
+      reason: reasons.reverse().join(', '),
+    },
+  });   
+  return {
+    event,
+    reason: reasons.reverse().join(', '),
+    success: false,
+  };
+};
+// #endregion                          MAIN PAGE EVENT CHECK
+
+// #region [rgba(0, 180, 0, 0.1)]      SINGLE PAGE EVENT CHECK
+// #endregion                          SINGLE PAGE EVENT CHECK
+
+// #region [rgba(0, 240, 0, 0.1)]      MAIN PAGE
+scraper.mainPage = async function () {
+  const availableBaseEvents = await this.checkBaseEventAvailable(workerData.family);
+  if (availableBaseEvents) {
+    const thisWorkersEvents = availableBaseEvents.filter(
+      (eventEl, index) => index % workerData.workerCount === workerData.index,
+    );
+    return this.mainPageEnd({ stopFunctie: null, rawEvents: thisWorkersEvents });
+  }
+
+  const { stopFunctie, page } = await this.mainPageStart();
+
+  await this.waitTime(50);
+
+  const cookiesNodig = await page.evaluate(() =>
+    document.querySelector('html').classList.contains('has-open-cookie'),
+  );
+
+  if (cookiesNodig) {
+    await page.evaluate(() => {
+      document.querySelector("[name*='marketing']").click();
+      document.querySelector('.cookie__settings .cookie__process').click();
+    });
+    await this.waitTime(50);
+  }
+
```

Comment:
Dit is dus voor websites die 
1. zoals paradiso een heel lang programma hebben en
2. die dynamisch ingeladen wordt op bereiken onderkant pagina. DWZ on page load is nog niet alles daar.